workspaces/tests/parallel_transactions.rs

```diff
@@ -38,3 +40,42 @@ async fn test_parallel() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_parallel_async() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox().await?;
+    let contract = worker.dev_deploy(STATUS_MSG_CONTRACT).await?;
+    let account = worker.dev_create_account().await?;
+
+    // Create a queue statuses we can check the status of later.
+    let mut statuses = VecDeque::new();
+    for msg in ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"] {
+        let status = account
+            .call(contract.id(), "set_status")
+            .args_json(json!({
+                "message": msg,
+            }))
+            .transact_async()
+            .await?;
+        statuses.push_back(status);
+    }
+
+    // Retry checking the statuses of all transactions until the queue is empty
+    // with all transactions completed.
+    while let Some(status) = statuses.pop_front() {
+        if let Err(_err) = status.status().await {
+            statuses.push_back(status);
+        }
+    }
```

Comment:
Feels like it would be much more ergonomic if the type returned from `transact_async` was a future so that you can simply use something like `try_join_all`. Also somewhat confusing what this test is doing because there is no need for the polling to cycle if you are waiting on all futures to complete.

What was the reasoning for not using a future type for the `transact_async` return?

Reply:
yeah it could just be a `Future`, but I feel like other information like the crypto hash and account-id, someone might want to reference for themselves. But in that case, let's have this `TransactionStatus` impl `Future` then so we can get the same result